### PR TITLE
Option for choosing OpenCL

### PR DESCRIPTION
**Pull request by JohannesKarwou**
new option in the `config` file:

- `GPU = True`, openMM uses CUDA
- `GPU = False`, openMM uses CPU
- `GPU = OpenCL`, openMM uses OpenCL

### CODE REVIEW

File: transformato/analysis.py [hunk 1]
```diff
@@ -117,12 +117,25 @@ def _generate_openMM_system(self, env: str, lambda_state: int) -> Simulation:
         psf = CharmmPsfFile(psf_file_path)
 
         # generate simulations object and set states
-        if self.configuration["simulation"]["GPU"] == True:
+        if (
+            self.configuration["simulation"]["GPU"]
+            and self.configuration["simulation"]["GPU"] == True
```
Comment (wiederm) [inline transformato/analysis.py:122]: that double-check is redundant, right?

File: transformato/state.py [hunk 2]
```diff
@@ -422,14 +422,34 @@ def _change_platform(self, file):
         g = open(f"{file}_tmp", "w+")
         i = 0  # counting lines
 
-        if self.configuration["simulation"]["GPU"]:
+        if (
+            self.configuration["simulation"]["GPU"]
+            and self.configuration["simulation"]["GPU"] == True
```
Comment (wiederm) [inline transformato/state.py:427]: I think this is also redundant

**Review by wiederm — CHANGES_REQUESTED** [review]
I would prefer a routine that checks if `CUDA` is available and if it is, it will use `CUDA`, otherwise it will fall back to `openCL`.
Here are some examples in which people have done this before:
https://programtalk.com/python-examples/simtk.openmm.Platform.getPlatformByName/

**Comment by JohannesKarwou** [comment]
> I would prefer a routine that checks if `CUDA` is available and if it is, it will use `CUDA`, otherwise it will fall back to `openCL`. Here are some examples in which people have done this before: https://programtalk.com/python-examples/simtk.openmm.Platform.getPlatformByName/

**Comment by JohannesKarwou** [comment]
There is an option already implemented by CHARMM-GUI I think we can use that, I will try it soon

```
# Set platform
DEFAULT_PLATFORMS = "CUDA", "OpenCL", "CPU"
enabled_platforms = [
    Platform.getPlatform(i).getName() for i in range(Platform.getNumPlatforms())
]
if args.platform:
    if not args.platform[0] in enabled_platforms:
        print(
            "Unable to find OpenMM platform '{}'; exiting".format(args.platform[0]),
            file=sys.stderr,
        )
        sys.exit(1)

    platform = Platform.getPlatformByName(args.platform[0])
else:
    for platform in DEFAULT_PLATFORMS:
        if platform in enabled_platforms:
            platform = Platform.getPlatformByName(platform)
            break
    if isinstance(platform, str):
        print(
            "Unable to find any OpenMM platform; exiting".format(args.platform[0]),
            file=sys.stderr,
        )
        sys.exit(1)
```

**Pull request closed without merging**